docs/src/main/docs/cdi-extensions.adoc

```diff
@@ -0,0 +1,221 @@
+
+   1. Clone the Oracle Cloud Infrastructure Java SDK artifacts.
+   
+      [source,bash]
+      ----
+      git clone --depth 1 --branch v1.2.44 https://github.com/oracle/oci-java-sdk.git
+      ----
+      
+   2. Install the Oracle Cloud Infrastructure Java SDK artifacts in your local Maven repository.
+   
+      [source,bash]
+      ----
+      mvn -B -U -f oci-java-sdk.git/pom.xml \
+          -Dmaven.test.skip=true \
+          -Dmaven.source.skip=true \
+          -Dmaven.javadoc.skip=true \
+          -Dlombok.delombok.skip=true \
+          -pl bmc-objectstorage \
+          -am \
+          install
+      ----  
+   
+   3. Specify the following dependency in your application's Maven POM:
+   
+      [source,xml]
+      ----
+      <dependency>
+         <groupId>io.helidon.integrations.cdi</groupId>
+         <artifactId>helidon-integrations-cdi-oci-objectstorage</artifactId>
+         <version>1.0.0</version>
+         <scope>runtime</scope>
+      </dependency>
+      ----
+
+== Injecting an object using a CDI extension
+
+The following are examples of field-injection and constructor-injection for each of the Helicon CDI extensions:
+ 
+=== Injecting a HikariCP data source
+
+The following examples show how to create a `DataSource` named `orders` in your application.
+
+* Field-injection example:
+
+[source,java]
+----
+@Inject
+@Named("orders")
+private DataSource ordersDataSource;
+----
+
+* Constructor-injection example:
+
+[source,java]
+----
+private final DataSource ds;
+  
+@Inject
+public YourConstructor(@Named("orders") DataSource ds) {
+  super();
+  this.ds = ds;
+}
+----
+
+The extension implements this injection point by creating an https://static.javadoc.io/com.zaxxer/HikariCP/2.7.8/com/zaxxer/hikari/HikariDataSource.html[HikariDataSource] object in the http://docs.jboss.org/cdi/api/2.0/javax/enterprise/context/ApplicationScoped.html[application scope].
+
+To locate the configuration of the object, the extension uses the https://static.javadoc.io/org.eclipse.microprofile.config/microprofile-config-api/1.3/index.html?overview-summary.html[MicroProfile
+Config API].
+
+https://static.javadoc.io/org.eclipse.microprofile.config/microprofile-config-api/1.3/org/eclipse/microprofile/config/Config.html#getPropertyNames--[Property
+names] that start with `javax.sql.DataSource.`_dataSourceName_`.` are parsed, and the remaining portion of each name is treated
+as a https://github.com/brettwooldridge/HikariCP/blob/dev/README.md#configuration-knobs-baby[Hikari
+connection pool property].
+
+For example, a property with the name `javax.sql.DataSource.orders.dataSourceClassName` and value `org.h2.jdbcx.JdbcDataSource` together together with other similarly-named properties results in a Hikari connection pool with the `dataSourceClassName` property set to `org.h2.jdbcx.JdbcDataSource`.
+  
+=== Injecting a Jedis client
+
+The following examples show how to create an Jedis pool named `orders` in your application code.
```

Comment:
Typo: "create an Jedis pool": create a Jedis pool.

Also, perhaps "inject" instead of "create"?  Or maybe "create and inject"?

Reply:
Changed to "create and inject a Jedis pool"